src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -158,6 +158,9 @@ private IVsMonitorSelection VsMonitorSelection
             _dteEvents = _dte.Events.DTEEvents;
             _dteEvents.OnBeginShutdown += OnBeginShutDown;
 
+            // Set the JoinableTaskFactory for the current version of visual studio.
+            NuGetUIThreadHelper.SetJoinableTaskFactoryFromService(ServiceLocator.PackageServiceProvider);
```

Comment:
don't use `ServiceLocator`. `PackageServiceProvider` is just `NuGetPackage` so simply pass `this`.
And also `componentModel` is fetched already, consider passing it instead.